options/m_option.h

```diff
@@ -424,63 +424,63 @@ struct m_option {
-#define UPDATE_LAVFI_COMPLEX    (1 << 18) // --lavfi-complex
-#define UPDATE_HWDEC            (1 << 20) // --hwdec
-#define UPDATE_DVB_PROG         (1 << 21) // some --dvbin-...
-#define UPDATE_SUB_HARD         (1 << 22) // subtitle opts. that need full reinit
-#define UPDATE_SUB_EXTS         (1 << 23) // update internal list of sub exts
-#define UPDATE_VIDEO            (1 << 24) // force redraw if needed
-#define UPDATE_VO               (1 << 25) // reinit the VO
-#define UPDATE_CLIPBOARD        (1 << 26) // reinit the clipboard
-#define UPDATE_OPT_LAST         (1 << 26)
+#define UPDATE_OPT_FIRST        (1ULL << 8)
+#define UPDATE_TERM             (1ULL << 8)  // terminal options
+#define UPDATE_SUB_FILT         (1ULL << 9)  // subtitle filter options
+#define UPDATE_OSD              (1ULL << 10) // related to OSD rendering
+#define UPDATE_BUILTIN_SCRIPTS  (1ULL << 11) // osc/ytdl/stats
+#define UPDATE_IMGPAR           (1ULL << 12) // video image params overrides
+#define UPDATE_INPUT            (1ULL << 13) // mostly --input-* options
+#define UPDATE_AUDIO            (1ULL << 14) // --audio-channels etc.
+#define UPDATE_PRIORITY         (1ULL << 15) // --priority (Windows-only)
+#define UPDATE_SCREENSAVER      (1ULL << 16) // --stop-screensaver
+#define UPDATE_VOL              (1ULL << 17) // softvol related options
+#define UPDATE_LAVFI_COMPLEX    (1ULL << 18) // --lavfi-complex
+#define UPDATE_HWDEC            (1ULL << 20) // --hwdec
+#define UPDATE_DVB_PROG         (1ULL << 21) // some --dvbin-...
+#define UPDATE_SUB_HARD         (1ULL << 22) // subtitle opts. that need full reinit
+#define UPDATE_SUB_EXTS         (1ULL << 23) // update internal list of sub exts
+#define UPDATE_VIDEO            (1ULL << 24) // force redraw if needed
+#define UPDATE_VO               (1ULL << 25) // reinit the VO
+#define UPDATE_CLIPBOARD        (1ULL << 26) // reinit the clipboard
+#define UPDATE_OPT_LAST         (1ULL << 26)
 
 // All bits between _FIRST and _LAST (inclusive)
 #define UPDATE_OPTS_MASK \
     (((UPDATE_OPT_LAST << 1) - 1) & ~(unsigned)(UPDATE_OPT_FIRST - 1))
 
 // type_float/type_double: string "default" is parsed as NaN (and reverse)
-#define M_OPT_DEFAULT_NAN       (1 << 26)
+#define M_OPT_DEFAULT_NAN       (1ULL << 31)
 
 // type time: string "no" maps to MP_NOPTS_VALUE (if unset, NOPTS is rejected)
 // and
 // parsing: "--no-opt" is parsed as "--opt=no"
-#define M_OPT_ALLOW_NO          (1 << 27)
+#define M_OPT_ALLOW_NO          (1ULL << 32)
 
 // type channels: disallow "auto" (still accept ""), limit list to at most 1 item.
-#define M_OPT_CHANNELS_LIMITED  (1 << 28)
+#define M_OPT_CHANNELS_LIMITED  (1ULL << 33)
 
 // type_float/type_double: controls if pretty print should trim trailing zeros
-#define M_OPT_FIXED_LEN_PRINT   (1 << 29)
+#define M_OPT_FIXED_LEN_PRINT   (1ULL << 34)
 
 // Like M_OPT_TYPE_OPTIONAL_PARAM.
-#define M_OPT_OPTIONAL_PARAM    (1 << 30)
+#define M_OPT_OPTIONAL_PARAM    (1ULL << 35)
```

Comment:
why not start these from the end (63) instead?

Reply:
That's clever.